Spigot-Server-Patches/0397-Call-player-spectator-target-events.patch

```diff
@@ -27,30 +27,33 @@ index 7ac07ac07ac0..7ac07ac07ac0 100644
  
 -        this.cv = (Entity) (entity == null ? this : entity);
 -        if (entity1 != this.cv) {
+-            this.playerConnection.sendPacket(new PacketPlayOutCamera(this.cv));
+-            this.playerConnection.a(this.cv.locX, this.cv.locY, this.cv.locZ, this.yaw, this.pitch, TeleportCause.SPECTATE); // CraftBukkit
 +        if (newSpectatorTarget == null) {
 +            newSpectatorTarget = this;
-+        }
+         }
+ 
++        if (entity1 == newSpectatorTarget) return; // new spec target is the current spec target
 +
-+        if (entity1 != newSpectatorTarget) {
-+            if (newSpectatorTarget == this) {
-+                com.destroystokyo.paper.event.player.PlayerStopSpectatingEntityEvent playerStopSpectatingEntityEvent = new com.destroystokyo.paper.event.player.PlayerStopSpectatingEntityEvent(this.getBukkitEntity(), entity1.getBukkitEntity());
++        if (newSpectatorTarget == this) {
++            com.destroystokyo.paper.event.player.PlayerStopSpectatingEntityEvent playerStopSpectatingEntityEvent = new com.destroystokyo.paper.event.player.PlayerStopSpectatingEntityEvent(this.getBukkitEntity(), entity1.getBukkitEntity());
 +
-+                if (!playerStopSpectatingEntityEvent.callEvent()) {
-+                    return;
-+                }
-+            } else {
-+                com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent playerStartSpectatingEntityEvent = new com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent(this.getBukkitEntity(), entity1.getBukkitEntity(), newSpectatorTarget.getBukkitEntity());
++            if (!playerStopSpectatingEntityEvent.callEvent()) {
++                return;
++            }
++        } else {
++            com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent playerStartSpectatingEntityEvent = new com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent(this.getBukkitEntity(), entity1.getBukkitEntity(), newSpectatorTarget.getBukkitEntity());
 +
-+                if (!playerStartSpectatingEntityEvent.callEvent()) {
-+                    return;
-+                }
++            if (!playerStartSpectatingEntityEvent.callEvent()) {
++                return;
 +            }
++        }
 +
-             this.playerConnection.sendPacket(new PacketPlayOutCamera(this.cv));
-             this.playerConnection.a(this.cv.locX, this.cv.locY, this.cv.locZ, this.yaw, this.pitch, TeleportCause.SPECTATE); // CraftBukkit
-         }
- 
 +        setSpectatorTargetField(newSpectatorTarget);
++
++        this.playerConnection.sendPacket(new PacketPlayOutCamera(newSpectatorTarget));
++        this.playerConnection.a(this.cv.locX, this.cv.locY, this.cv.locZ, this.yaw, this.pitch, TeleportCause.SPECTATE); // CraftBukkit
```

Comment:
Still no `// Paper` tags on your actual changes

Reply:
Adding it now